File: Ruby/civitas/pkg/civitas-0.0.1/lib/casilla_juez.rb
#encoding: utf­8
require_relative "casilla.rb"
module Civitas
  
  class CasillaJuez < Casilla
    
    def initialize(carcel,_nombre)
       super(_nombre)
      @@carcel = carcel
     
    end
    
    def recibe_jugador(iactual,todos)
      if(jugador_correcto(iactual, todos))
        informe(iactual, todos)
        todos.at(iactual).encarcelar(@@carcel)
      end
    end
    
    def to_s
      return "Te han pillado involucrado en los ERES vas a la carcel"
    end
  end
end

File: Java/civitas/src/civitas/CasillaJuez.java
package civitas;

import java.util.ArrayList;

public class CasillaJuez extends Casilla{
    private static int carcel;
    
    CasillaJuez(int numCasilla, String _nombre){
        super(_nombre);
        carcel = numCasilla;
    }
    
    @Override
    public void recibeJugador(int iActual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(iActual, todos)){
            super.informe(iActual, todos);
            todos.get(iActual).encarcelar(carcel);
        }
    }
    
    @Override
    public String toString(){
        return "Te han pillado involucrado en los ERES vas la carcel " + Integer.toString(carcel);
    }
}
